repo_policy_compliance/github_client.py

```diff
@@ -34,24 +34,46 @@ def wrapper(*args: P.args, **kwargs: P.kwargs) -> R:
 
         Args:
             args: The positional arguments passed to the method, github_client is prepended when
-            calling the wrapped method
+                calling the wrapped method
             kwargs: The keywords arguments passed to the method
 
         Raises:
             InputError: If the GitHub token environment variable is not provided or empty.
+            GithubClientError: If the Github client encountered an error.
 
         Returns:
             The return value after calling the wrapped function with the injected GitHub client.
 
         """
-        github_token = os.getenv(GITHUB_TOKEN_ENV_NAME)
-        if not github_token:
-            raise InputError(
-                f"The {GITHUB_TOKEN_ENV_NAME} environment variable was not provided or empty, it "
-                f"is needed for interactions with GitHub, got: {GITHUB_TOKEN_ENV_NAME!r}"
-            )
-        github_client = Github(login_or_token=github_token)
-
-        return func(github_client, *args, **kwargs)
+        if len(args) < 3 and "github_client" not in kwargs:
```

Comment:
What is the reason for checking the `len` of `args`? Some if the functions will have varying numbers of `args`, like `execute_job` will have 3. Also, as currently implemented, the arguments can be passed as either an arg or kwarg depending on user preference (e.g., both `target_branch_protection("repository 1", "branch 1")` and `target_branch_protection(repository_name="repository 1", branch_name="branch 1")` is valid, in the first one they will be passed in `args`, in the second in `kwargs`